src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2587,32 +2566,41 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             {
                 resultType = node.Type.SetUnknownNullabilityForReferenceTypes();
                 resultState = NullableFlowState.NotNull;
-            }
-            else
-            {
-                var resultTypeWithAnnotations = TypeWithAnnotations.Create(resultType);
-                TypeWithState convertedConsequenceResult = default;
-                TypeWithState convertedAlternativeResult = default;
 
-                if (consequenceEndReachable)
+                var resultTypeWithState = TypeWithState.Create(resultType, resultState);
+                var visitResult = new VisitResult(resultTypeWithState, resultTypeWithState.ToTypeWithAnnotations());
+
+                if (consequence != node.Consequence)
                 {
-                    convertedConsequenceResult = convertResult(
-                        node.Consequence,
-                        consequence,
-                        consequenceConversion,
-                        resultTypeWithAnnotations,
-                        consequenceRValue);
+                    Debug.Assert(node.Consequence.Kind == BoundKind.Conversion);
+                    SetAnalyzedNullability(node.Consequence, visitResult);
                 }
 
-                if (alternativeEndReachable)
+                if (alternative != node.Alternative)
                 {
-                    convertedAlternativeResult = convertResult(
-                        node.Alternative,
-                        alternative,
-                        alternativeConversion,
-                        resultTypeWithAnnotations,
-                        alternativeRValue);
+                    Debug.Assert(node.Alternative.Kind == BoundKind.Conversion);
+                    SetAnalyzedNullability(node.Alternative, visitResult);
```

Comment:
>SetAnalyzedNullability(node.Alternative, visitResult); [](start = 20, length = 54)

I was debating whether the consequence and the alternative should get different visit results. But I think this is correct (shared visit result).